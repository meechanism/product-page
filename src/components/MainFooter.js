import React from 'react';
import styled from 'styled-components';

import Text from '../components/Text';

const FooterWrapper = styled.footer`
  padding: 2rem;
  text-align: center;
`;

const Footer = () => (
  <FooterWrapper>
    <Text>Â© 2020 Elmwest. All Rights Reserved.</Text>
    <Text>
      <a href="/">Terms and Conditions</a> | <a href="/">Privacy Policy</a> |{' '}
      <a href="/">Legal Statement</a>
    </Text>
  </FooterWrapper>
);

export default Footer;
