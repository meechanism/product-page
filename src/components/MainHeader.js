import React from 'react';
import styled from 'styled-components';
import { Link } from 'react-router-dom';

import Colors from '../styles/colors';
import MediaQuery from '../styles/media-queries';

export const Branding = styled.span`
  display: inline-block;
  font-weight: bold;
  font-size: 3rem;
  border-left: 1rem solid ${Colors.secondary};
  padding-left: 2rem;

  ${MediaQuery.medium`
    font-size: 1.5rem;
  `}
`;

export const HeaderWrapper = styled.div`
  display: flex;
  padding: 1.5rem 1.5rem 1.5rem 0;
  justify-content: space-between;
  align-items: center;
  border-bottom: 1px solid ${Colors.gray};

  ${MediaQuery.medium`
    padding: 1rem 1rem 1rem 0;
  `}
`;

export const NavList = styled.ul`
  list-style-type: none;
  margin: 0;
  padding: 0;
`;

export const NavItem = styled.li`
  margin: 0;
  padding: 0;
  display: inline-block;

  a {
    color: ${Colors.black};
    padding: 1.5rem;
    text-transform: uppercase;
    &:hover {
      color: ${Colors.primary};
    }

    ${MediaQuery.medium`
      padding: 1rem;
  `}
  }
`;

const MainHeader = () => (
  <HeaderWrapper>
    <Branding>Elmwest</Branding>
    <nav>
      <NavList>
        <NavItem>
          <Link to="/">Home</Link>
        </NavItem>
        <NavItem>
          <Link to={'/'}>Shop</Link>
        </NavItem>
        <NavItem>
          <Link to={'/'}>Bedroom</Link>
        </NavItem>
      </NavList>
    </nav>
  </HeaderWrapper>
);

export default MainHeader;
