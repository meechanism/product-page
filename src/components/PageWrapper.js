import React from 'react';
import styled from 'styled-components';
import GlobalStyles from '../styles/GlobalStyles';

import MainHeader from './MainHeader';
import MainFooter from './MainFooter';

const Body = styled.main`
  max-width: 960px;
  margin: 2rem auto;
  min-height: 50vh;
  padding: 0 2rem;
`;

export default ({ children }) => (
  <>
    <GlobalStyles />
    <MainHeader />
    <Body>{children}</Body>
    <MainFooter />
  </>
);
