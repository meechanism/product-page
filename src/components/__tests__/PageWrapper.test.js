import React from 'react';
import PageWrapper from '../PageWrapper';
import MainFooter from '../MainFooter';
import MainHeader from '../MainHeader';
import GlobalStyles from '../../styles/GlobalStyles';

describe('PageWrapper component', () => {
  it('should render', () => {
    const output = shallow(
      <PageWrapper>
        <p>test</p>
      </PageWrapper>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });

  it('should contain main header, fooder, and global styles', () => {
    const output = shallow(
      <PageWrapper>
        <p>test</p>
      </PageWrapper>
    );
    expect(output.find(MainHeader).exists()).toBeTruthy();
    expect(output.find(MainFooter).exists()).toBeTruthy();
    expect(output.find(GlobalStyles).exists()).toBeTruthy();
  });
});
