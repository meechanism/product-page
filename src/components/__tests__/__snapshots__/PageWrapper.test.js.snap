// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWrapper component should render 1`] = `
<Fragment>
  <Memo(GlobalStyleComponent) />
  <MainHeader />
  <styled.main>
    <p>
      test
    </p>
  </styled.main>
  <Footer />
</Fragment>
`;
